includes/wizards/class-setup-wizard.php

```diff
@@ -0,0 +1,144 @@
+
+/**
+ * Setup Newspack.
+ */
+class Setup_Wizard extends Wizard {
+	/**
+	 * The slug of this wizard.
+	 *
+	 * @var string
+	 */
+	protected $slug = 'newspack-setup-wizard';
+	/**
+	 * The capability required to access this wizard.
+	 *
+	 * @var string
+	 */
+	protected $capability = 'edit_products';
+
+	/**
+	 * Display a link to this wizard in the Newspack submenu.
+	 *
+	 * @var bool
+	 */
+	protected $hidden = false;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		parent::__construct();
+		add_action( 'current_screen', [ $this, 'redirect_to_setup' ] );
+		add_action( 'rest_api_init', [ $this, 'register_api_endpoints' ] );
+	}
+
+	/**
+	 * Get the name for this wizard.
+	 *
+	 * @return string The wizard name.
+	 */
+	public function get_name() {
+		return esc_html__( 'Setup', 'newspack' );
+	}
+	/**
+	 * Get the description of this wizard.
+	 *
+	 * @return string The wizard description.
+	 */
+	public function get_description() {
+		return esc_html__( 'Setup Newspack.', 'newspack' );
+	}
+	/**
+	 * Get the expected duration of this wizard.
+	 *
+	 * @return string The wizard length.
+	 */
+	public function get_length() {
+		return esc_html__( '10 minutes', 'newspack' );
+	}
+
+	/**
+	 * Register the endpoints needed for the wizard screens.
+	 */
+	public function register_api_endpoints() {
+		// Update option when setup is complete.
+		register_rest_route(
+			'newspack/v1/wizard/' . $this->slug,
+			'/complete',
+			[
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => [ $this, 'api_complete' ],
+				'permission_callback' => [ $this, 'api_permissions_check' ],
+			]
+		);
+	}
+
+	/**
+	 * Update option when setup is complete.
+	 *
+	 * @return WP_REST_Response containing info.
+	 */
+	public function api_complete() {
+		update_option( NEWSPACK_SETUP_COMPLETE, true );
+		return rest_ensure_response( [ 'complete' => true ] );
+	}
+
+	/**
+	 * Enqueue Subscriptions Wizard scripts and styles.
+	 */
+	public function enqueue_scripts_and_styles() {
+		parent::enqueue_scripts_and_styles();
+		wp_enqueue_media();
+		if ( filter_input( INPUT_GET, 'page', FILTER_SANITIZE_STRING ) !== $this->slug ) {
+			return;
+		}
+		wp_enqueue_script(
+			'newspack-setup-wizard',
+			Newspack::plugin_url() . '/assets/dist/setup.js',
+			[ 'wp-components' ],
+			filemtime( dirname( NEWSPACK_PLUGIN_FILE ) . '/assets/dist/setup.js' ),
+			true
+		);
+		wp_register_style(
+			'newspack-setup-wizard',
+			Newspack::plugin_url() . '/assets/dist/setup.css',
+			[ 'wp-components' ],
+			filemtime( dirname( NEWSPACK_PLUGIN_FILE ) . '/assets/dist/setup.css' )
+		);
+		wp_style_add_data( 'newspack-setup-wizard', 'rtl', 'replace' );
+		wp_enqueue_style( 'newspack-setup-wizard' );
+	}
+
+	/**
+	 * If initial setup is incomplete, redirect to Setup Wizard.
+	 */
+	public function redirect_to_setup() {
+		if ( get_option( NEWSPACK_SETUP_COMPLETE ) ) {
+			return;
+		}
+		try {
+			$screen = get_current_screen();
```

Comment:
`get_current_screen` will never throw an exception. You can [see the source here](https://developer.wordpress.org/reference/functions/get_current_screen/) (it's only like 6 lines).

Reply:
Removed in https://github.com/Automattic/newspack-plugin/pull/114/commits/1d9b54030ad995992a90a1cdc8d224079aded128

I did this only because of this scary line in this docs: https://codex.wordpress.org/Function_Reference/get_current_screen

```
This function is defined on most admin pages, but not all. Thus there are cases where is_admin() will return true, but attempting to call get_current_screen() will result in a fatal error because it is not defined. One known example is wp-admin/customize.php.
```